middleware/kubernetes/kubernetes.go

```diff
@@ -270,32 +303,71 @@ func (k *Kubernetes) getRecordsForServiceItems(serviceItems []*api.Service, zone
 }
 
 // Get performs the call to the Kubernetes http API.
-func (k *Kubernetes) Get(namespace string, nsWildcard bool, servicename string, serviceWildcard bool, typeName string) ([]*api.Service, error) {
+func (k *Kubernetes) Get(namespace string, servicename string, endpointname string, port string, protocol string, typeName string) (services []service, err error) {
 	switch {
 	case typeName == "pod":
-		return nil, fmt.Errorf("pod not implemented")
+		return nil, fmt.Errorf("%v not implemented", typeName)
 	default:
-		return k.getServices(namespace, nsWildcard, servicename, serviceWildcard)
+		return k.getServices(namespace, servicename, endpointname, port, protocol)
 	}
 }
 
-func (k *Kubernetes) getServices(namespace string, nsWildcard bool, servicename string, serviceWildcard bool) ([]*api.Service, error) {
+func (k *Kubernetes) getServices(namespace string, servicename string, endpointname string, port string, protocol string) ([]service, error) {
```

Comment:
nit: you don't need to repeat `string` for every parameter, only needed for the last one.
`namespace, servicename, endpointname, port, protocol string